src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/IntermediateDateTypeConfiguratorTest.java

```diff
@@ -0,0 +1,602 @@
+            .value(stringValue("priorityIntermediateDate,nextHearingDate,dueDate"))
+            .build();
+
+
+        ConfigurationDmnEvaluationResponse nextHearingDate = ConfigurationDmnEvaluationResponse.builder()
+            .name(stringValue("nextHearingDate"))
+            .value(stringValue(NEXT_HEARING_DATE_VALUE))
+            .build();
+
+        String intermediateDateName = "nextHearingDuration";
+        ConfigurationDmnEvaluationResponse nextHearingDurationOrigin = ConfigurationDmnEvaluationResponse.builder()
+            .name(stringValue(intermediateDateName + ORIGIN_SUFFIX))
+            .value(stringValue(INTERMEDIATE_DATE_VALUE))
+            .build();
+
+        ConfigurationDmnEvaluationResponse priorityIntermediateDate = ConfigurationDmnEvaluationResponse.builder()
+            .name(stringValue("priorityIntermediateDate"))
+            .value(stringValue(GIVEN_DATE.plusDays(2).format(DateTimeFormatter.ofPattern("yyyy-MM-dd")) + "T16:00"))
+            .build();
+        List<ConfigurationDmnEvaluationResponse> evaluationResponses = readDueAndPriorityDateOriginFields(
+            intermediateDateName, dueDateOriginEarliest, priorityDateOriginEarliest, calculatedDates,
+            nextHearingDate, nextHearingDurationOrigin, priorityIntermediateDate
+        );
+
+        List<ConfigurationDmnEvaluationResponse> configurationDmnEvaluationResponses
+            = dateTypeConfigurator.configureDates(evaluationResponses, false, false);
+
+        assertThat(configurationDmnEvaluationResponses)
+            .hasSize(6)
+            .isEqualTo(List.of(
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(stringValue("calculatedDates"))
+                    .value(stringValue(
+                        "nextHearingDate,nextHearingDuration,dueDate,priorityIntermediateDate,priorityDate"))
+                    .build(),
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(stringValue("nextHearingDate"))
+                    .value(stringValue("2022-10-13T16:00"))
+                    .build(),
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(stringValue("nextHearingDuration"))
+                    .value(stringValue("2022-10-19T21:00"))
+                    .build(),
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(stringValue("dueDate"))
+                    .value(stringValue("2022-10-18T17:00"))
+                    .build(),
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(stringValue("priorityIntermediateDate"))
+                    .value(stringValue("2022-10-15T16:00"))
+                    .build(),
+                ConfigurationDmnEvaluationResponse.builder()
+                    .name(stringValue("priorityDate"))
+                    .value(stringValue("2022-10-21T21:00"))
+                    .build()
+            ));
+    }
+
```

Comment:
What about AC1 & AC2? i.e.

1. No 'date' or  ..Origin' or ...'OriginReq' or '....OriginEarliest' or '..OriginLatest' have been defined for the intermediate date **then intermediate date can not be calculated and the value of empty is captured for that date**.
2. No  'date'  BUT either 'OriginReq' or '....OriginEarliest' or '..OriginLatest {}have been defined and refer to a calculated date which is empty/NULL then then intermediate date can not be calculated and the value of empty is captured for that date.